client.go

```diff
@@ -2376,6 +2377,8 @@ func (c *pipelineConnClient) DoDeadline(req *Request, resp *Response, deadline t
 	}
 
 	w := acquirePipelineWork(&c.workPool, timeout)
+	defer releasePipelineWork(&c.workPool, w)
```

Comment:
This will go wrong in the last ErrTimeout case where w might still be in use by the worker.

Reply:
You're right, reverted it.

Reply:
But still, we need to handle this case and make sure to put it back into the pool, I'll send another PR for it later.